__test__/ClusterTSnePlot.test.js

```diff
@@ -113,6 +113,18 @@ describe(`ClusterTSnePlot`, () => {
     expect(wrapper.find(PlotSettingsDropdown).length).toBe(2)
   })
 
+  test(`contains no PlotSettingsDropdowns when flag is false`, () => {
+    const onChangeColourBy = () => {}
+    const onChangePerplexity = () => {}
+    const plotData = {
+      series: []
+    }
+
+    const wrapper = mount(<ClusterTSnePlot height={500} ks={[]} metadata={[]} selectedColourBy={`0`} onChangeColourBy={onChangeColourBy} perplexities={[]} selectedPerplexity={0} onChangePerplexity={onChangePerplexity} loading={true} plotData={plotData} showControls={false}/>)
```

Comment:
I see the props are repeated in each test, which is not from this branch, but could you please refactor a bit with a const `props` which contains all the shared ones?

Reply:
Refactor the code to share the common props.

Reply:
Actually what I mean is the props of `height`, `ks`, `metadata` etc shared by all wrappers. You could assign them as a variable and pass it to `ClusterTSnePlot` to reduce the repeat of codes. Same comment for `GeneExpressionTSnePlot.test`. Please remember to add `showControls` props explicitly in the test.

Reply:
Right. Could you please try to put all common props into one variable like
```
const props = {height: 500, ks:...., plotData:..., metadata:...}
```
and then pass the `props` into each tested component and add any additional props if needed?

Reply:
Implemented as suggested.